server/src/transactions/transactions.controller.ts

```diff
@@ -1,40 +1,72 @@
-import { Controller, Post, Body, Get, Query, Param, Delete, Patch } from '@nestjs/common';
+import {
+  Body,
+  Controller,
+  Delete,
+  Get,
+  Param,
+  Patch,
+  Post,
+  Query,
+  Request,
+  UnauthorizedException,
+  UseGuards,
+} from '@nestjs/common';
 
 import { TransactionsService } from './transactions.service';
 import { CreateTransactionDto } from './dto/create-transaction.dto';
 import { Transaction } from './entities/transaction.entity';
 import { GetTransactionsDto } from './dto/get-transactions-filter.dto';
 import { UpdateTransactionDto } from './dto/update-transaction.dto';
+import { CookieAuthGuard } from '../auth/guards/cookie-auth.guard';
+import { UsersService } from '../users/users.service';
+
+import type { AuthedRequest } from '../types';
 
 @Controller('transactions')
 export class TransactionsController {
-  constructor(private transactionsService: TransactionsService) {}
+  constructor(
+    private transactionsService: TransactionsService,
+    private usersService: UsersService,
+  ) {}
 
+  @UseGuards(CookieAuthGuard)
   @Post()
-  create(@Body() createTransactionDto: CreateTransactionDto): Promise<Transaction> {
+  async create(@Body() createTransactionDto: CreateTransactionDto): Promise<Transaction> {
     return this.transactionsService.createTransaction(createTransactionDto);
   }
 
+  @UseGuards(CookieAuthGuard)
   @Get()
-  get(@Query() getTransactionsDto: GetTransactionsDto): Promise<Transaction[]> {
+  async get(@Query() getTransactionsDto: GetTransactionsDto): Promise<Transaction[]> {
     return this.transactionsService.getTransactions(getTransactionsDto);
   }
 
+  @UseGuards(CookieAuthGuard)
+  @Get('/currentUser')
```

Comment:
Optional:
- web routes don't differentiate between `currentuser` or `currentUser`, if you want, change it to: `current-user`.

Reply:
actually going to opt to just use the existing `Get('/')` endpoint and pass user_id in as part of the query params

Reply:
nvm, went with current-user